File: 1213-intersection-of-three-sorted-arrays/1213-intersection-of-three-sorted-arrays.py
class Solution:
    def arraysIntersection(self, arr1: List[int], arr2: List[int], arr3: List[int]) -> List[int]:
        def looping(arr: List[int]) -> None:
            for num in arr:
                counting_sort[num] += 1
        counting_sort = [0] * 2001
        looping(arr1)
        looping(arr2)
        looping(arr3)
        
        ans = []
        for i in range(1, 2001):
            if counting_sort[i] == 3:
                ans.append(i)
        
        return ans 

File: 1213-intersection-of-three-sorted-arrays/1213-intersection-of-three-sorted-arrays.cpp
class Solution {
public:
    vector<int> arraysIntersection(vector<int>& arr1, vector<int>& arr2, vector<int>& arr3) {
        vector<int> ans;
        map<int, int> counter;
        
        // Concatenate all vectors together
        for (int num : arr1) ++counter[num];
        for (int num : arr2) ++counter[num];
        for (int num : arr3) ++counter[num];
        
        for (auto& [num, freq] : counter){
            if (freq == 3) ans.push_back(num);
        }
        return ans;
    }
};